pages/settings/account/index.tsx

```diff
@@ -271,7 +272,15 @@ const AccountSettings = () => {
   const loading = userInfoLoading || status === 'loading'
   const { username, name, discordUsername } = data?.userInfo?.user || {}
 
-  if (loading || !data) {
+  // prevents the page UI from showing if unauthenticated
+  if (loading || status === 'unauthenticated') {
+    if (status === 'unauthenticated') {
```

Comment:
Is the `unauthenticated` status something that other pages will use as well? If so, then it should be made a constant

Reply:
You repeat this logic twice, put it into a variable:

```
status === 'unauthenticated'
```

Reply:
Actually, I notice you're only saving one line of code(`return <LoadingSpinner />`), but the result is that it makes the code more complex and harder to maintain since you are coupling logic together, and also using nested if statements as well. Nested ifs increase the number of linearly independent paths we have, which makes the codebase harder to understand and maintain

Reply:
Lets decouple the logic and have something like this:

```

if (loading) {
    return <LoadingSpinner />
}
if (status === 'unauthenticated') {
  router.push({
    pathname: '/login',
    query: { next: router.asPath }
  })
  return <LoadingSpinner /> 
}
```

Reply:
Also, if other places in the codebase are using loadingSpinner when redirecting a user, then a function should be made to reproduce this functionality of redirecting + showing spinner

Reply:
What does showing a spinner do when redirecting a user? How long does a user see the spinner for when they get redirected?

Reply:
> What does showing a spinner do when redirecting a user? Does the user see the spinner for when they are getting redirected?

It prevents the UI (forms, buttons,..etc) from showing if the user isn't yet determined to be unauthenticated or not. If you tried to go to your profile and refresh the page multiple times, you'll notice the loading spinner disappears and the UI shows up with dummy data because it didn't yet fetch your data from the GraphQL API. This is what that line is trying to achieve, display a loading spinner until all the data that's necessary for the page is fetched.

This apparently isn't the best approach, so we're looking into [this feature](#2624).